lib/system/gc_regions.nim

```diff
@@ -102,13 +102,13 @@ var
   tlRegion {.threadVar.}: MemRegion
 #  tempStrRegion {.threadVar.}: MemRegion  # not yet used
 
-template withRegion*(r: MemRegion; body: untyped) =
+template withRegion*(r: var MemRegion; body: untyped) =
```

Comment:
It's a template, ``r`` is mutable. (Caused by the substitution mechanism of templates.)

Reply:
Well changing the signature gives a better error message upon using `withRegion` on an immutable `MemRegion`:

```
test.nim(19, 11) Error: type mismatch: got <MemRegion, void>
but expected one of:
template withRegion(r: var MemRegion; body: untyped)
  first type mismatch at position: 1
  required type for r: var MemRegion
  but expression 'mr' is immutable, not 'var'
```

vs

```
test.nim(19, 12) Error: 'mr' cannot be assigned to
```